chain/network/src/types.rs

```diff
@@ -1,1050 +1,10 @@
-    Ip(IpAddr),
-    IpPort(SocketAddr),
-}
-
-impl PatternAddr {
-    pub fn contains(&self, addr: &SocketAddr) -> bool {
-        match self {
-            PatternAddr::Ip(pattern) => &addr.ip() == pattern,
-            PatternAddr::IpPort(pattern) => addr == pattern,
-        }
-    }
-}
-
-impl FromStr for PatternAddr {
-    type Err = AddrParseError;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if let Ok(pattern) = s.parse::<IpAddr>() {
-            return Ok(PatternAddr::Ip(pattern));
-        }
-
-        s.parse::<SocketAddr>().map(PatternAddr::IpPort)
-    }
-}
-
-/// Status of the known peers.
-#[derive(BorshSerialize, BorshDeserialize, Serialize, Eq, PartialEq, Debug, Clone)]
-pub enum KnownPeerStatus {
-    Unknown,
-    NotConnected,
-    Connected,
-    Banned(ReasonForBan, u64),
-}
-
-impl KnownPeerStatus {
-    pub fn is_banned(&self) -> bool {
-        match self {
-            KnownPeerStatus::Banned(_, _) => true,
-            _ => false,
-        }
-    }
-}
-
-/// Information node stores about known peers.
-#[derive(BorshSerialize, BorshDeserialize, Debug, Clone)]
-pub struct KnownPeerState {
-    pub peer_info: PeerInfo,
-    pub status: KnownPeerStatus,
-    pub first_seen: u64,
-    pub last_seen: u64,
-}
-
-impl KnownPeerState {
-    pub fn new(peer_info: PeerInfo) -> Self {
-        KnownPeerState {
-            peer_info,
-            status: KnownPeerStatus::Unknown,
-            first_seen: to_timestamp(Utc::now()),
-            last_seen: to_timestamp(Utc::now()),
-        }
-    }
-
-    pub fn first_seen(&self) -> DateTime<Utc> {
-        from_timestamp(self.first_seen)
-    }
-
-    pub fn last_seen(&self) -> DateTime<Utc> {
-        from_timestamp(self.last_seen)
-    }
-}
-
-impl TryFrom<Vec<u8>> for KnownPeerState {
-    type Error = Box<dyn std::error::Error>;
-
-    fn try_from(bytes: Vec<u8>) -> Result<KnownPeerState, Self::Error> {
-        KnownPeerState::try_from_slice(&bytes).map_err(|err| err.into())
-    }
-}
-
-/// Actor message that holds the TCP stream from an inbound TCP connection
-#[derive(Message)]
-#[rtype(result = "()")]
-pub struct InboundTcpConnect {
-    /// Tcp stream of the inbound connections
-    pub stream: TcpStream,
-}
-
-impl InboundTcpConnect {
-    /// Method to create a new InboundTcpConnect message from a TCP stream
-    pub fn new(stream: TcpStream) -> InboundTcpConnect {
-        InboundTcpConnect { stream }
-    }
-}
-
-/// Actor message to request the creation of an outbound TCP connection to a peer.
-#[derive(Message)]
-#[rtype(result = "()")]
-pub struct OutboundTcpConnect {
-    /// Peer information of the outbound connection
-    pub peer_info: PeerInfo,
-}
-
-#[derive(Message, Clone, Debug)]
-#[rtype(result = "()")]
-pub struct SendMessage {
-    pub message: PeerMessage,
-}
+pub use near_network_primitives::types::*;
 
 /// Actor message to consolidate potential new peer.
 /// Returns if connection should be kept or dropped.
```

Comment:
I feel that all the Actor message types should be part of -primitives. Does `Consolidate` use something that blocks us from moving it to the primitives?

Reply:
Yeah, it does...

1. [`near_network::types::Consolidate`](https://github.com/near/nearcore/blob/a44be206018b547f53cb3b704f3a822e0d9d5db0/chain/network/src/types.rs#L1051-L1052)

  ```rust
  pub struct Consolidate { // `Consolidate` wasn't extracted because 👇🏽
      pub actor: Addr<Peer>, // <-- `Peer` wasn't extracted, because 👇🏽
  ...
  }
  ```

2. [`near_network::peer::Peer`](https://github.com/near/nearcore/blob/a44be206018b547f53cb3b704f3a822e0d9d5db0/chain/network/src/peer.rs#L173)

  ```rust
  pub struct Peer {
  ...
      peer_manager_addr: Addr<PeerManagerActor>, // <-- `PeerManagerActor` wasn't extracted, because 👇🏽
  ...
  }
  ```

3. [`near_network::peer_manager::PeerManagerActor`](https://github.com/near/nearcore/blob/a44be206018b547f53cb3b704f3a822e0d9d5db0/chain/network/src/peer_manager.rs#L158)

  ```rust
  pub struct PeerManagerActor {
  ...
      peer_store: PeerStore, // `PeerStore` wasn't extracted, because 👇🏽
  ...
  }
  ```

4. [`near_network::peer_store::PeerStore`](https://github.com/near/nearcore/blob/a44be206018b547f53cb3b704f3a822e0d9d5db0/chain/network/src/peer_store.rs#L48-L49)

  ```rust
  pub struct PeerStore {
      store: Arc<Store>, // `Store` wasn't extracted, because 👇🏽
  ...
  }
  ```

5. [`near_store::PeerStore`](https://github.com/near/nearcore/blob/a44be206018b547f53cb3b704f3a822e0d9d5db0/core/store/src/lib.rs#L48-L49)

  ```rust
  pub struct PeerStore {
      storage: Pin<Arc<dyn Database>>, // 👈🏽 rocksdb
  }
  ```

Reply:
I see. Sounds reasonable to me. Thanks for the detailed explanation :+1: